src/glmfit.jl

```diff
@@ -174,6 +174,13 @@ confint(obj::AbstractGLM) = confint(obj, 0.95)
 
 deviance(m::AbstractGLM) = deviance(m.rr)
 
+function perfectlyseparable(m::AbstractGLM)
+    y_pred = m.rr.y
+    y = m.pp.X[:,1]
+    order = sortperm(y_pred)
+    count(diff(y[order]).!=0) == 1  #perfectly separable? (Given only 1 class interface in score-based sort) (bool)
```

Comment:
Turn this into an explicit loop to avoid an allocation. That will also allow returning early as soon as you find two consecutive elements that aren't equal.

Reply:
Ok. Updated